### PR TITLE
Volunteering system

### PR DESCRIPTION
**Pull request by abdelq**
This is a volunteering system for wafer that stemmed from the need of one for DebConf17.
It's inspired by [jrial/fosdem-volunteers](https://github.com/jrial/fosdem-volunteers)

I haven't used Django in a long time, so a thorough review would be appreciated.

Thanks to @olasd for the various improvements.

Notes:
* Mass generation of video tasks is suboptimal.

### CODE REVIEW

**Comment by hodgestar** [comment]
This looks interesting! Shout when you're done committing and we can give it a thorough review?

File: wafer/volunteers/views.py [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+
+        if self.object in volunteer.tasks.all():
+            volunteer.tasks.remove(self.object)
+            self.object.volunteers.remove(volunteer)
+        elif self.object.nbr_volunteers() < self.object.nbr_volunteers_max:
+            volunteer.tasks.add(self.object)
+            self.object.volunteers.add(volunteer)
+
+        return redirect('wafer_task', pk=self.object.pk)
+
```
Comment (drnlm) [inline wafer/volunteers/views.py]: Is it useful to have the tasks viewable without authenticating?

If it is, should the tasks be included in the static site generation?

**Comment by abdelq** [comment]
@drnlm I haven't pushed the rest of the changes yet, since I was busy with the conference.

**Comment by hodgestar** [comment]
@abdelq How is this looking? :)

**Comment by abdelq** [comment]
@hodgestar Been busy.
I just pushed some final changes and rebased. I think it's good to start reviewing this PR now.

**Pull request closed without merging**